HelperFunctions.ps1

```diff
@@ -1,1121 +1,1171 @@
+    Param(
+        [string] $containerName,
+        [switch] $includeTestLibrariesOnly,
+        [switch] $includeTestFrameworkOnly,
+        [switch] $includeTestRunnerOnly,
+        [switch] $includePerformanceToolkit
+    )
+
+    Invoke-ScriptInBCContainer -containerName $containerName -scriptblock { Param($includeTestLibrariesOnly, $includeTestFrameworkOnly, $includeTestRunnerOnly, $includePerformanceToolkit)
+    
+        $version = [Version](Get-Item "C:\Program Files\Microsoft Dynamics NAV\*\Service\Microsoft.Dynamics.Nav.Server.exe").VersionInfo.FileVersion
+
+        # Add Test Framework
+        $apps = @()
+        if (($version -ge [Version]"19.0.0.0") -and (Test-Path 'C:\Applications\TestFramework\TestLibraries\permissions mock')) {
+            $apps += @(get-childitem -Path "C:\Applications\TestFramework\TestLibraries\permissions mock\*.*" -recurse -filter "*.app")
+        }
+        $apps += @(get-childitem -Path "C:\Applications\TestFramework\TestRunner\*.*" -recurse -filter "*.app")
+
+        if (!$includeTestRunnerOnly) {
+            $apps += @(get-childitem -Path "C:\Applications\TestFramework\TestLibraries\*.*" -recurse -filter "*.app")
+
+            if (!$includeTestFrameworkOnly) {
+                # Add Test Libraries
+                $apps += "Microsoft_System Application Test Library.app", "Microsoft_Tests-TestLibraries.app" | ForEach-Object {
+                    @(get-childitem -Path "C:\Applications\*.*" -recurse -filter $_)
+                }
+    
+                if (!$includeTestLibrariesOnly) {
+    
+                    # Add Tests
+                    $apps += @(get-childitem -Path "C:\Applications\*.*" -recurse -filter "Microsoft_Tests-*.app") | Where-Object { $_ -notlike "*\Microsoft_Tests-TestLibraries.app" -and ($version.Major -ge 17 -or ($_ -notlike "*\Microsoft_Tests-Marketing.app")) -and $_ -notlike "*\Microsoft_Tests-SINGLESERVER.app" }
+                }
+            }
+        }
+
+        if ($includePerformanceToolkit) {
+            $apps += @(get-childitem -Path "C:\Applications\TestFramework\PerformanceToolkit\*.*" -recurse -filter "*Toolkit.app")
+            if (!$includeTestFrameworkOnly) {
+                $apps += @(get-childitem -Path "C:\Applications\TestFramework\PerformanceToolkit\*.*" -recurse -filter "*.app" -exclude "*Toolkit.app")
+            }
+        }
+
+        $apps | ForEach-Object {
+            $appFile = Get-ChildItem -path "c:\applications.*\*.*" -recurse -filter ($_.Name).Replace(".app", "_*.app")
+            if (!($appFile)) {
+                $appFile = $_
+            }
+            $appFile.FullName
+        }
+    } -argumentList $includeTestLibrariesOnly, $includeTestFrameworkOnly, $includeTestRunnerOnly, $includePerformanceToolkit
+}
+
+function GetExtendedErrorMessage {
+    Param(
+        $errorRecord
+    )
+
+    $exception = $errorRecord.Exception
+    $message = $exception.Message
+
+    if ($errorRecord.ErrorDetails) {
```

Comment:
This will fail if ErrorDetails isn't a property, please move it inside the try / catch